301-expressroute-circuit-vnet-connection/azureDeploy.json

```diff
@@ -0,0 +1,175 @@
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the virtual network to create."
+            }
+        },
+        "addressPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "The address space in CIDR notation for the new virtual network."
+            }
+        },
+        "subnet1Name": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the first subnet in the new virtual network."
+            }
+        },
+        "gatewaySubnet": {
+            "type": "string",
+            "defaultValue": "GatewaySubnet",
+            "allowedValues": [
+                "GatewaySubnet"
+            ],
+            "metadata": {
+                "description": "The name of the subnet where Gateway is to be deployed. This must always be named GatewaySubnet."
+            }
+        },
+        "subnet1Prefix": {
+            "type": "string",
+            "metadata": {
+                "description": "The address range in CIDR notation for the first subnet."
+            }
+        },
+        "gatewaySubnetPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "The address range in CIDR notation for the Gateway subnet. For ExpressRoute enabled Gateways, this must be minimum of /28."
+            }
+        },
+        "gatewayPublicIPName": {
+            "type": "string",
+            "metadata": {
+                "description": "The resource name given to the public IP attached to the gateway."
+            }
+        },
+        "gatewayName": {
+            "type": "string",
+            "metadata": {
+                "description": "The resource name given to the ExpressRoute gateway."
+            }
+        },
+        "connectionName": {
+            "type": "string",
+            "metadata": {
+                "description": "The resource name given to the Connection which links VNet Gateway to ExpressRoute circuit."
+            }
+        },
+        "circuitName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the ExpressRoute circuit with which the VNet Gateway needs to connect. The Circuit must be already created successfully and must have its circuitProvisioningState property set to 'Enabled', and serviceProviderProvisioningState property set to 'Provisioned'. The Circuit must also have a BGP Peering of type AzurePrivatePeering."
+            }
+        }
+    },
+    "variables": {
+        "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',parameters('virtualNetworkName'))]",
+        "gatewaySubnetRef": "[concat(variables('vnetID'),'/subnets/',parameters('gatewaySubnet'))]"
+    },
+    "resources": [
+        {
+            "apiVersion": "2015-05-01-preview",
```

Comment:
It's a good idea to make the api version a variable to make it easier to move to newer api versions. Also can you update to 2015-06-15 ?